Search in settings

KnollFrank opened the pull request:
My first attempt for ticket "Search in settings #13973"

RZR-UA commented:
Please follow original formatting to avoid the huge amount of changes lines: +1,598 −1,490

KnollFrank commented:
> Please follow original formatting to avoid the huge amount of changes lines: +1,598 −1,490

Where ist the formatter I can use in Android Studio, please?

vshcherb commented:
Probably it's tab vs spaces, we use tabs by default

KnollFrank commented:
> Please follow original formatting to avoid the huge amount of changes lines: +1,598 −1,490

ok, I changed the formatting from spaces to tabs.

franco999 commented:
Anxious to see and test how this great new feature progresses. 
Excellent work, I hope it can be merged soon to the main project 👍

mnalis commented on OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:
```diff
@@ -26,8 +26,8 @@
 import net.osmand.ResultMatcher;
 import net.osmand.binary.BinaryMapIndexReader;
-import net.osmand.binary.BinaryMapIndexReader.SearchPoiTypeFilter;
 import net.osmand.binary.BinaryMapIndexReader.SearchPoiAdditionalFilter;
+import net.osmand.binary.BinaryMapIndexReader.SearchPoiTypeFilter;
```
I've noticed that there a lot superficial changes  (probably done by your IDE) which only change the orders or imports. This is just one example, see https://github.com/osmandapp/OsmAnd/pull/20016/files?diff=split&w=1 for full list.

Those enlarge the diff needlessly, and make it harder to review changes. Would it be possible that you clean those up?

KnollFrank replied:
Hi, the merge conflicts are resolved, imports reorganized and the apk is here: https://github.com/KnollFrank/OsmAnd/releases/download/testing/OsmAnd-nightlyFree-legacy-fat-debug.apk

mnalis reviewed (commented):
There seems to a dozen or so merge conflicts preventing the merge of this PR; could you resolve those (i.e. base the PR on latest master so it applies cleanly)?

Also, I think it would help if you'd build an nightly `.apk` with your PR included (and publish it on your github), so interested people in https://github.com/osmandapp/OsmAnd/issues/13973 could try it out and verify how it works and provide feedback (there are a lot of users willing to beta-test and few that know hot to build the app).

When those are confirmed working ok, I hope core developers will merge the PR to master, so wider community can be exposed to it.

And thanks a lot on working on this much needed improvement!

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding two FIXMEs

KnollFrank committed:
adapting a FIXME

KnollFrank committed:
fixing: 1. search for "Afrikaans", 2. click search result for Skiing => crash

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
updating FIXMEs

KnollFrank committed:
Fixing: 1. search for "Afrikaans", 2. click search result to open dialog, 3. click Afrikaans, 4. click Apply to close the dialog, then the following item erroneously does not show "af" (for Afrikaans) in it's description, but displays the "old" description.

KnollFrank committed:
formatting and refactoring

KnollFrank committed:
Fixing: description of "Road Style" item is not updated after search (e.g. american road) and property change

KnollFrank committed:
refactoring

KnollFrank committed:
adding yet failing test SettingsSearchIntegrationTest

KnollFrank committed:
formatting

KnollFrank committed:
adding some TODOs

KnollFrank committed:
refactoring

KnollFrank committed:
making SettingsSearchIntegrationTest pass

KnollFrank committed:
refactoring

KnollFrank committed:
adding test AdaptMapStyleAfterSearchTest

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding test AdaptRoadStyleAfterSearchTest

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding yet failing test SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_Hide_HouseNumbers()

KnollFrank committed:
introducing MultiSelectionDialogFragment

KnollFrank committed:
refactoring

KnollFrank committed:
making SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_Hide_HouseNumbers() pass

KnollFrank committed:
making search results for "Hide" dialog clickable

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding some TODOs

KnollFrank committed:
refactoring

KnollFrank committed:
adding yet failing test SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapMode_sunrise()

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
making test shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapMode_sunrise() pass

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
making MapMode search results clickable

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding yet failing test SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapSource_OfflineVectorMaps()

KnollFrank committed:
commenting

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
making test SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapSource_OfflineVectorMaps() pass^

KnollFrank committed:
making SettingsSearchTest pass again

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding tests shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapSource_onlineTiles_onlineMapsPluginActive() and shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapSource_onlineTiles_onlineMapsPluginInactive()

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
removing a superfluous FK-FIXME

KnollFrank committed:
disabling showToastMessage() for "map source" when building search database

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding yet failing test SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapSource_nonInstalledTileSource()

KnollFrank committed:
making test SettingsSearchTest.shouldSearchAndFind_ConfigureScreenFragment_ConfigureMap_MapSource_nonInstalledTileSource() pass

KnollFrank committed:
adding a TODO

KnollFrank committed:
adapting some TODOs

KnollFrank committed:
removing a done FK-FIXME

KnollFrank committed:
adding a FK-FIXME

KnollFrank committed:
adding TileSourceTemplatesProvider with cache for performance when building search database

KnollFrank committed:
refactoring

KnollFrank committed:
updating a TODO

KnollFrank committed:
adding a FIXME

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
fixing:
1. navigate to InstallMapLayersDialogFragment (Settings -> Driving -> Configure map -> Map source... -> Add more...)
2. select "Top Yandex RU"
3. click Apply
Then "Top Yandex RU" appears in MapLayerSelectionDialogFragment (Settings -> Driving -> Configure map -> Map source...) but this new entry of MapLayerSelectionDialogFragment is not yet stored in the search database and hence can't be found via a search query.

KnollFrank committed:
formatting

KnollFrank committed:
using sqlite version of settings search library

KnollFrank committed:
using new version of settings search library

KnollFrank committed:
using new version of settings search library

KnollFrank committed:
using new version of settings search library and new version of guava

KnollFrank committed:
adapting to new version of settings search library

KnollFrank commented:
> @KnollFrank can you share what is the status of PR for this amazing feature? Any prognosis when it might be ready for merging?
> 
> I would really love to see this land in OsmAnd!

Hello @mnalis, I won't have access to my computer again until Monday to answer your question.

KnollFrank committed:
creating a prepackaged search database file

KnollFrank committed:
Revert "creating a prepackaged search database file"

This reverts commit 60a1d7289e0a5abdedec392d5ad5c93d38c8dddf.

KnollFrank committed:
merging build files from master

KnollFrank committed:
updating some dependencies

KnollFrank committed:
formatting

KnollFrank committed:
starting to precompute searchable preferences database

KnollFrank committed:
formatting

KnollFrank committed:
adding Configuration in settings database

KnollFrank committed:
adapting a comment

KnollFrank committed:
adapting to new version of settings library

KnollFrank committed:
adapting to new version of settings library

KnollFrank committed:
adding gradle task generatePrepackagedDatabase

KnollFrank committed:
renaming gradle task generatePrepackagedDatabase to generatePrepackagedPreferencesDatabase

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
commenting

KnollFrank committed:
formatting

KnollFrank committed:
refactoring

KnollFrank committed:
argh!!!

KnollFrank committed:
uninstalling App before executing gradle task generateAndPullPreferencesDatabase

KnollFrank committed:
refactoring

KnollFrank committed:
formatting

KnollFrank committed:
refactoring

KnollFrank committed:
making more tests pass again

KnollFrank committed:
making even more tests pass again

KnollFrank committed:
removing a TODO

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding InstallMapSourceThenSearchSettingsTest

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
formatting

KnollFrank committed:
refactoring

KnollFrank committed:
refining InstallMapSourceThenSearchSettingsTest

KnollFrank committed:
continuing to make InstallMapLayersDialogFragmentFactory work

KnollFrank committed:
making InstallMapSourceThenSearchSettingsTest pass again

KnollFrank committed:
refactoring

KnollFrank committed:
refining InstallMapSourceThenSearchSettingsTest

KnollFrank committed:
refactoring

KnollFrank committed:
refining InstallMapSourceThenSearchSettingsTest

KnollFrank committed:
refactoring

KnollFrank committed:
refining SearchDatabaseRootedAtConfigureMapFragmentAdapter

KnollFrank committed:
refactoring

KnollFrank committed:
formatting

KnollFrank committed:
refactoring

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
removing increaseTimeout() and resetTimeout() in InstallMapSourceThenSearchSettingsTest

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
enableCacheForDownloadedTileSourceTemplatesWhileBuildingSearchDatabase() while applyGraphProcessorsToGraphs()

KnollFrank committed:
updating adapting to new version of settings search library with improved performance

KnollFrank committed:
adding SearchDatabaseRootedAtConfigureProfileFragmentAdapter

KnollFrank committed:
introducing SearchDatabaseRootedAtPreferenceFragmentAdapter

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
formatting

KnollFrank committed:
introducing PluginSettingsOfConfigureProfileFragmentAdapter

KnollFrank committed:
PluginSettingsOfConfigureProfileFragmentAdapter.addPluginEdgesOfConfigureProfileFragmentToGraph()

KnollFrank committed:
refactoring

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adding some FIXMEs

KnollFrank committed:
adding a TODO

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
fixing some tests

KnollFrank committed:
removing PluginSettingsOfConfigureProfileFragmentAdapter

KnollFrank committed:
refactoring

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
removing jgrapht dependency

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adding PluginSettingsOfConfigureProfileFragmentAdapter

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
refactoring

KnollFrank committed:
adapting a comment

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
refactoring

KnollFrank committed:
refining PluginSettingsOfConfigureProfileFragmentAdapter

KnollFrank committed:
argh!!!!

KnollFrank committed:
formatting

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
using PluginSettingsOfConfigureProfileFragmentAdapter

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
making SettingsSearchTest (almost) pass again

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
formatting

KnollFrank committed:
improving performance when searching for preferences a little bit

KnollFrank committed:
removing some done TODOs

KnollFrank committed:
fixing "en-" is too special

KnollFrank committed:
removing BaseSettingsFragment.configureSettingsSearch

KnollFrank committed:
removing PrepareShow

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
fixing hard coded string bug

KnollFrank committed:
fixing gradle task generateAndPullPreferencesDatabase

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding testLocale param

KnollFrank committed:
refactoring

KnollFrank committed:
removing parameter generatePreferencesDb

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adding gradle task generateDatabasesForLocales

KnollFrank committed:
refactoring

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
making DatabaseSourceUrl configurable

KnollFrank committed:
refactoring

KnollFrank committed:
hiding ProgressBar when building search database

KnollFrank committed:
adding getAllConfiguredLocales()

KnollFrank committed:
refactoring

KnollFrank committed:
adding startEmulator and stopEmulator around processing each locale for the search database

KnollFrank committed:
fixing getAllConfiguredLocales()

KnollFrank committed:
renaming testLocale to locale

KnollFrank committed:
adapting to new version of settings search library

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
adapting DATABASE_URL_TEMPLATE for release

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
refactoring

KnollFrank committed:
removing an unfeasible TODO
